PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageOne.java

```diff
@@ -0,0 +1,118 @@
+package com.microsoft.azuretools.hdinsight.projects;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Group;
+import org.scalaide.ui.ScalaImages;
+import org.scalaide.ui.internal.wizards.NewScalaProjectWizardPageOne;
+
+import com.microsoft.azuretools.adauth.StringUtils;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+
+class HDInsightScalaPageOne extends NewScalaProjectWizardPageOne {
+	
+	private SparkLibraryOptionsPanel sparkLibraryOptionsPanel;
+	
+	@Override
+	public boolean canFlipToNextPage() {
+		final String jarPathString = sparkLibraryOptionsPanel.getSparkLibraryPath();
+		if(StringUtils.isNullOrEmpty(jarPathString)) {
+			return false;
+		}
+		return super.canFlipToNextPage();
+	}
+	
+	@Override
+	public IClasspathEntry[] getDefaultClasspathEntries() {
+		final IClasspathEntry[] entries = super.getDefaultClasspathEntries();
+
+		final IClasspathEntry[] newEntries = new IClasspathEntry[entries.length + 1];
+
+		Display.getDefault().syncExec(new Runnable() {
+			@Override
+			public void run() {
+				final String jarPathString = sparkLibraryOptionsPanel.getSparkLibraryPath();
+				if (StringUtils.isNullOrEmpty(jarPathString)) {
+					DefaultLoader.getUIHelper().showError("Spark Library Path cannot be null",
+							"Spark Project Settings");
+				} else {
+					IPath jarPath = new Path(jarPathString);
+					IClasspathEntry sparkEntry = JavaCore.newLibraryEntry(jarPath, null, null);
+					System.arraycopy(entries, 0, newEntries, 0, entries.length);
```

Comment:
Yes


---
In reply to: [132867205](https://github.com/Microsoft/azure-tools-for-java/pull/884#discussion_r132867205) [](ancestors = 132867205,132864101)